src/PROPOSAL/detail/PROPOSAL/particle/Particle.cxx

```diff
@@ -125,10 +126,20 @@ double ParticleState::GetMomentum() const
 StochasticLoss::StochasticLoss(int type, double loss_energy, const Vector3D& position,
                                const Vector3D& direction, double time,
                                double propagated_distance,
-                               double parent_particle_energy)
+                               double parent_particle_energy, size_t target_hash)
                                : Loss(type, loss_energy, parent_particle_energy),
                                position(position), direction(direction), time(time),
-                               propagated_distance(propagated_distance) {}
+                               propagated_distance(propagated_distance),
+                               target_hash(target_hash) {}
+
+Component StochasticLoss::GetComponent() const {
```

Comment:
How to get the medium?

Reply:
I added a `medium_map` to the `Medium` class to get the corresponding Medium to a hash.
The `GetComponentForHash()` method has also been moved to be a static function of the Component class, there is also a `GetMediumForHash()` method now in the Medium class.

I think for the future, we should use pointers instead of these global maps.

Reply:
> I think for the future, we should use pointers instead of these global maps.

well, then create an issue